### PR TITLE
[16.0][OU-ADD] website_payment: Migration to 16.0

### PR DESCRIPTION
**Pull request by victoralmau**
Migration to 16.0

@Tecnativa TT43633

### CODE REVIEW

**Comment by legalsylvain** [comment]
/ocabot migration website_payment

<dependency></dependency>

**Review by chienandalu — APPROVED** [review]

File: openupgrade_scripts/scripts/website_payment/16.0.1.0/post-migration.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.delete_records_safely_by_xml_id(
+        env,
+        [
+            "website_payment.mail_template_donation",
+        ],
+    )
+    openupgrade.load_data(
+        env.cr, "website_payment", "migrations/16.0.1.0/noupdate_changes.xml"
```
Comment (remytms) [inline openupgrade_scripts/scripts/website_payment/16.0.1.0/post-migration.py]: I think that the correct location is `"16.0.1.0/noupdate_changes.xml"` and not under the `migrations` directory.

File: openupgrade_scripts/scripts/website_payment/16.0.1.0/post-migration.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.delete_records_safely_by_xml_id(
+        env,
+        [
+            "website_payment.mail_template_donation",
+        ],
+    )
```
Comment (remytms) [inline openupgrade_scripts/scripts/website_payment/16.0.1.0/post-migration.py]: I think that simply loading `noupdate_changes.xml` is enough. No need to delete this record before.

Based on: http://oca.github.io/OpenUpgrade/use_cases/noupdate_xml_entry_changed.html

**Review by remytms — CHANGES_REQUESTED** [review]
I may be wrong, but I think that this PR needs some changes.

File: openupgrade_scripts/scripts/website_payment/16.0.1.0/post-migration.py (hunk 2 above)
Comment (pedrobaeza) [reply openupgrade_scripts/scripts/website_payment/16.0.1.0/post-migration.py]: Mostly: you need to reset the translation for the field name.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
@victoralmau please rebase an attend comments.

And other thing: this is not "Migration to 16.0", but "Migration scripts"

**Comment by victoralmau** [comment]
Changes done.

File: openupgrade_scripts/scripts/website_payment/16.0.1.0/post-migration.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+_translations_to_delete = [
+    "mail_template_donation",
+]
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.load_data(env.cr, "website_payment", "16.0.1.0/noupdate_changes.xml")
+    openupgrade.delete_record_translations(env.cr, "load_data", _translations_to_delete)
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/website_payment/16.0.1.0/post-migration.py]: Please limit the deletion to the name field.

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
For getting green CI:

/ocabot merge nobump

**Pull request merged**